UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentAncillaryFiles.cs

```diff
@@ -0,0 +1,78 @@
+        private readonly IFileSystemHelper fileSystemHelper;
+
+        public FulfilmentAncillaryFiles(ILogger<FulfilmentDataService> logger,
+                                        IOptions<FileShareServiceConfiguration> fileShareServiceConfig,
+                                        IFileSystemHelper fileSystemHelper)
+        {
+            this.logger = logger;
+            this.fileShareServiceConfig = fileShareServiceConfig;
+            this.fileSystemHelper = fileSystemHelper;
+        }
+        public bool CreateProductFile(string batchId, string exchangeSetInfoPath, string correlationId, SalesCatalogueDataResponse salesCatalogueDataResponse)
+        {
+            if (salesCatalogueDataResponse.ResponseCode == HttpStatusCode.OK)
+            {
+                string fileName = fileShareServiceConfig.Value.ProductFileName;
+                string file = Path.Combine(exchangeSetInfoPath, fileName);
+
+                var productsBuilder = new ProductListBuilder();
+                foreach (var product in salesCatalogueDataResponse.ResponseBody.OrderBy(p => p.ProductName))
+                    productsBuilder.Add(new ProductListEntry()
+                    {
+                        ProductName = product.ProductName + fileShareServiceConfig.Value.BaseCellExtension,
+                        Compression = product.Compression,
+                        Encryption = product.Encryption,
+                        BaseCellIssueDate = product.BaseCellIssueDate,
+                        BaseCellEdition = product.BaseCellEditionNumber,
+                        IssueDateLatestUpdate = product.IssueDateLatestUpdate,
+                        LatestUpdateNumber = product.LatestUpdateNumber,
+                        FileSize = product.FileSize,
+                        TenDataCoverageCoordinates = ",,,,,,,,,,,,,,,,,,,",
+                        CellLimitSouthernmostLatitude = Convert.ToDecimal(product.CellLimitSouthernmostLatitude.ToString(Convert.ToString("0.0000000000"))),
+                        CellLimitWesternmostLatitude = Convert.ToDecimal(product.CellLimitWesternmostLatitude.ToString(Convert.ToString("0.0000000000"))),
+                        CellLimitNorthernmostLatitude = Convert.ToDecimal(product.CellLimitNorthernmostLatitude.ToString(Convert.ToString("0.0000000000"))),
+                        CellLimitEasternmostLatitude = Convert.ToDecimal(product.CellLimitEasternmostLatitude.ToString(Convert.ToString("0.0000000000"))),
+                        BaseCellUpdateNumber = product.BaseCellUpdateNumber,
+                        LastUpdateNumberForPreviousEdition = product.LastUpdateNumberForPreviousEdition,
+                        BaseCellLocation = product.BaseCellLocation,
+                        CancelledCellReplacements = string.Empty,
+                    });
+
+                var content = productsBuilder.WriteProductsList(DateTime.UtcNow);
+                var productFileName = file;
```

Comment:
Kindy remove this assignment. You can directly pass file in method fileSystemHelper.CreateFileContent(file, content).

Reply:
As per the Comment i have removed productFileName assignment.